paimon-core/src/main/java/org/apache/paimon/operation/LocalOrphanFilesClean.java

```diff
@@ -114,7 +118,7 @@ private List<String> getUsedFiles(String branch) {
         ManifestFile manifestFile =
                 table.switchToBranch(branch).store().manifestFileFactory().create();
         try {
-            List<String> manifests = new ArrayList<>();
+            Set<String> manifests = new HashSet<>();
```

Comment:
While reading multiple snapshots, there will be duplicate manifests. Deduplication can improve efficiency.